render/host.go

```diff
@@ -7,10 +7,7 @@ import (
 // HostRenderer is a Renderer which produces a renderable host
 // graph from the host topology.
 var HostRenderer = MakeReduce(
-	MakeMap(
-		MapEndpoint2Host,
-		EndpointRenderer,
-	),
+	endpoints2Hosts{},
```

Comment:
Unlike the `MakeMap` this replaces, this is not `Memoise`d. Which got me thinking about memoisation (again) and I have filed #2923 as a result.

Reply:
Yes, I couldn't see when memoisation would be useful for this particular case.
I did unify two calls to the `r` Renderer in `ConnectionJoin`, but that's trivial.